tests/bluetooth/bsim_bt/bsim_test_app/src/test_connect1.c

```diff
@@ -268,8 +275,7 @@ static void device_found(const bt_addr_le_t *addr, s8_t rssi, u8_t type,
 
 	/* We're only interested in connectable events */
 	if (type == BT_LE_ADV_IND || type == BT_LE_ADV_DIRECT_IND) {
-		/* TODO: Move this to a place it can be shared */
-		ad_parse(ad, eir_found, (void *)addr);
+		bt_data_parse(ad, eir_found, (void *)addr);
```

Comment:
This refactoring to use bt_data_parse() seems unrelated to the rest of the patch. Could you move that into a separate commit please? (can still be part of this same PR).

Reply:
Yes, I will do that.

Reply:
@jhedberg The commit has been split accordingly. Do you have any further comments?